R/set_salt.R

```diff
@@ -0,0 +1,36 @@
+
+#' @title Set Hash Salt
+#'
+#' @description
+#' This function sets a environment value to be used for hash-matched
+#' pseudo-random number generation (PRNG). This means that `salt`
+#' values are scoped.
+#'
+#' This is the conceptual equivalent of [set.seed()] with unmatched
+#' PRNG.
+#'
+#' @param salt an object that will be used as the starter object for
+#' all hash-matched. A plain integer is sufficient.
+#'
+#' @export
+set.salt <- function(salt) {
+  assign(".hash.salt", salt, envir = parent.frame())
```

Comment:
have you thought about doing this with options? I think it might be a lot more cran compliant

Reply:
A little bit, but the idea is to have a scope-driven salt value - that probably means setting something in an environment.

Reply:
but options can do that?

Reply:
or an environment variable (options -> environment variable would be the classic approach I think)

Reply:
this approach is an environment variable?

i don't think the options version let's you do this cleanly - unsetting the variable when leaving a scope seems messy?

Reply:
I don't think it is its a normal variable?

Reply:
This is what I mean (though I didn't mean via this nice little wrapper package): https://cran.r-project.org/web/packages/options/vignettes/envvars.html

Reply:
Like the way we handle the cache location in `epinowcast`: https://github.com/epinowcast/epinowcast/blob/12c24bf509704c4384902408f36dff86e73fef5b/R/model-tools.R#L635

Reply:
sorry, i should say: it's a normal variable, in the r-environment-scoping sense. so when r does its normal search enclosing scopes to find a token, it gets found. it also gets cleaned up etc like normal, & can exist in closer scope, while outer scope has its own (masked until that scope is returned to).